source/client/app/AppPlatform.cpp

```diff
@@ -210,7 +210,15 @@ bool AppPlatform::hasFileSystemAccess()
 
 std::string AppPlatform::getPatchData()
 {
-	return "";
+	AssetFile file = readAssetFile("patches/patch_data.txt");
+	std::string out = std::string(file.data, file.data + file.size);
+	delete file.data;
```

Comment:
Wouldn't that require `SoundDesc` maintaining a copy of `AssetFile` so the sound data isn't deleted?

Reply:
Yes. But you return `AssetFile` from `AppPlatform::readAssetFile` so you can just keep the `AssetFile` in memory (ideally you'd also use `std::move` to avoid copying)